athena-redshift/src/main/java/com/amazonaws/athena/connectors/redshift/RedshiftMetadataHandler.java

```diff
@@ -124,4 +128,15 @@ public GetDataSourceCapabilitiesResponse doGetDataSourceCapabilities(BlockAlloca
 
         return new GetDataSourceCapabilitiesResponse(request.getCatalogName(), capabilities.build());
     }
+
+    @Override
+    protected List<TableName> getPaginatedResults(Connection connection, String databaseName, int token, int limit) throws SQLException
```

Comment:
It works with materialized views. The previous implementation calls getMaterializedViews(), but it lets it error out because pg_catalog.pg_matviews is not supported by redshift. Pagination still keeps the essential part of the query.